source/Calamari.Aws/Commands/UploadAwsS3Command.cs

```diff
@@ -49,11 +49,10 @@ public override int Execute(string[] commandLineArguments)
             var substituter = new FileSubstituter(fileSystem);
             var bucketKeyProvider = new BucketKeyProvider();
             var targetType = GetTargetMode(targetMode);
-            var packageExtractor = new GenericPackageExtractorFactory().createStandardGenericPackageExtractor();
-
+            
             var conventions = new List<IConvention>
             {
-                new ExtractPackageToStagingDirectoryConvention(packageExtractor, fileSystem).When(_ => targetType == S3TargetMode.FileSelections),
+                new ExtractPackageToStagingDirectoryConvention(new CombinedPackageExtractor(), fileSystem).When(_ => targetType == S3TargetMode.FileSelections),
```

Comment:
Was going to inject it, but when we introduce `IConventionFactory` this will be centralized there anyway.